src/libraries/System.Net.Http/tests/UnitTests/Headers/CacheControlHeaderValueTest.cs

```diff
@@ -654,6 +654,27 @@ public void TryParse_SetOfInvalidValueStrings_ReturnsFalse()
             CheckInvalidTryParse("\u4F1A", 0);
         }
 
+        [Fact]
+        public void TryParseAndAddRawHeaderValue_TwoValidValues_NoInvalidValuesAdded()
+        {
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, "http://microsoft.com");
+
+            request.Headers.TryAddWithoutValidation(KnownHeaders.CacheControl.Descriptor, "min-fresh=123");
+            request.Headers.TryAddWithoutValidation(KnownHeaders.CacheControl.Descriptor, "max-stale");
+            object? storeValue = request.Headers.GetParsedAndInvalidValues(KnownHeaders.CacheControl.Descriptor);
```

Comment:
I think this test should be
```c#
TryAddWithoutValidation(validValue);
TryAddWithoutValidation(string.Empty);

Assert(values.Count == 1);
```

You can still have a list of values (one cache control + invalid ones), just not whitespace-only invalid values.

Reply:
Updated